src/collection.js

```diff
@@ -39,48 +39,55 @@ function makeItem (component, sources, props) {
   return newItem;
 }
 
-function Collection (component, sources = {}, handlers = {}, items = [], handler$Hash = {}, reducers = xs.create()) {
-  return {
-    add (additionalSources = {}) {
-      const newItem = makeItem(component, {...sources, ...additionalSources});
-      const handler$ = handlerStreams(component, newItem, handlers);
-      reducers.imitate(handler$);
-
-      return Collection(
-        component,
-        sources,
-        handlers,
-        [...items, newItem],
-        {
-          ...handler$Hash,
-          [newItem.id]: handler$
-        },
-        reducers
-      );
+function Collection (component, sources = {}, handlers = {}) {
+  const reducers = xs.create();
+  const proxy = {
+    next(value) {
+      reducers.shamefullySendNext(value);
     },
-
-    remove (itemForRemoval) {
-      const id = itemForRemoval && itemForRemoval.id;
-      id && handler$Hash[id] && handler$Hash[id].shamefullySendComplete();
-      return Collection(
-        component,
-        sources,
-        handlers,
-        items.filter(item => item !== itemForRemoval),
-        {
-          ...handler$Hash,
-          [id]: null
-        },  
-        reducers
-      );
-    },
-
-    asArray () {
-      return items.slice(); // returns a copy of items to avoid mutation
+    error(err) {
+      reducers.shamefullySendError(err);
     },
+    complete() {
+      // Maybe autoremove here?
+    }
+  }
 
-    reducers
-  };
+  return (function _Collection(items = [], handler$Hash = {}) {
```

Comment:
Yep, that makes sense as well
On 2/06/2016 6:44 PM, "Hypnosphi" <notifications@github.com> wrote:

> In src/collection.js
> <https://github.com/cyclejs/collection/pull/7#discussion_r65491141>:
>
> >
> > -    reducers
> > -  };
> > +  return (function _Collection(items = [], handler$Hash = {}) {
>
> I think I'd better make collection accept options as first argument which
> would be destructured as { component, sources, handlers, reducers, proxy }
> and leave items and handler$hash as separate variables
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/cyclejs/collection/pull/7/files/ede011b2be6382603ba51ee205f520345e171a2a#r65491141>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYUHbYwIQGxXCuVYT98sKED5pZ5Fobxks5qHnu9gaJpZM4IpbME>
> .
>